### PR TITLE
Internal #576: strptime strftime infinities

### PR DESCRIPTION
**Pull request by hawkfish**
Make it so we can round trip infinite temporal values through the datetime parser and formatter.
This now differs from PG but I think they are wrong.

fixes: duckdblabs/duckdb-internal#576

### CODE REVIEW

**Comment by hawkfish** [comment]
Rebasing required deleting the old branch and PR - sorry!

File: src/function/scalar/strftime_format.cpp [hunk 1]
```diff
@@ -733,7 +731,33 @@ bool StrpTimeFormat::Parse(string_t str, ParseResult &result) const {
 		data++;
 		size--;
 	}
+
+	//	Check for specials
 	idx_t pos = 0;
+	result.is_special = false;
+	if (Date::TryConvertDateSpecial(data, size, pos, Date::PINF)) {
```
Comment (Mytherin) [inline src/function/scalar/strftime_format.cpp:738]: This seems to reduce performance of `benchmark/micro/cast/strptime.benchmark` by around 10~15%, perhaps we can be a bit more clever in when we take this branch (e.g. check if the first character is a character?)

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! One comment:

File: src/function/scalar/strftime_format.cpp (hunk 1 above)
Comment (hawkfish) [reply src/function/scalar/strftime_format.cpp]: The benchmark is too fast for consistent results but profiling with Instruments showed a definite hot spot there.  Checking for size and the first character being alpha or minus removed the hot spot.

Comment (Mytherin) [reply src/function/scalar/strftime_format.cpp]: Great! When I ran it I increased the number of rows generated in the benchmark to double-check that it was not just noise

**Review by samansmink — APPROVED** [review]
LGTM!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**